packages/zk-wizard/src/zk-wizard.ts

```diff
@@ -240,12 +248,32 @@ export class ZkWizard {
         start}`,
     )
     // TODO handle genProof exception
-    let memo: Buffer | undefined
-    if (encryptTo !== undefined) {
-      const noteToEncrypt = tx.outflow.find(outflow =>
-        outflow.owner.eq(encryptTo),
-      )
-      if (noteToEncrypt instanceof Utxo) memo = noteToEncrypt.encrypt()
+    const encryptTo = option?.encryptTo
+    let memo: Memo | undefined
+    if (option?.memo === MemoVersion.V1) {
+      if (encryptTo !== undefined) {
+        const noteToEncrypt = tx.outflow.find(outflow =>
+          outflow.owner.eq(encryptTo),
+        )
+        if (noteToEncrypt instanceof Utxo) {
+          memo = {
```

Comment:
@wanseob If I'm sending a transaction and use a V1 memo, will I be able to discover the UTXO that I create for myself when the transaction is sent?

Reply:
Yes but you cannot discover the UTXO through calldata scan after resetting your wallet. To make it recoverable we have to use V2 memo. I implemented the default uses V2 & encrypt all utxo outputs